subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/m5/ToolingApiIdeaModelCrossVersionSpec.groovy

```diff
@@ -215,7 +215,16 @@ project(':impl') {
         lib.scope.scope == 'TEST'
 
         IdeaModuleDependency mod = libs.find {it instanceof IdeaModuleDependency}
-        mod.dependencyModule == project.modules.find { it.name == 'api'}
+
+        def currentVersion = GradleVersion.current()
+        if (currentVersion >= GradleVersion.version("3.1")) {
+            mod.targetModuleName == project.modules.find { it.name == 'api'}.getName()
+        }
+
+        if (currentVersion < GradleVersion.version("3.99")) {
+            mod.dependencyModule == project.modules.find { it.name == 'api'}
+        }
+
```

Comment:
@bmuschko We wouldn't want to preclude this testing in the event that 3.5.1 or if 3.6 came out for some reason.